### PR TITLE
Configure for Reducers, Actions

### PR DESCRIPTION
**Pull request by mnekx**
## Description

In this milestone, I configure Redux Store and write actions and reducer for adding and removing books

## Type of change
The following are the changes done

- [x] Add React-Redux (npm install react-redux).
- [x] Structure the application files using a "feature folder" approach and using the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for Redux files.
- [x] Write book's actions and reducer (in /src/redux/books/books.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).
- [x] Write categories actions and reducer (in /src/redux/categories/categories.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).
- [x] Configure the Redux Store (/src/redux/configureStore.js).

## Checklist:

- [x] No linter errors.
- [x] I have used the correct Gitflow.
- [x] Work is documented in a professional way.
- [x] List of JavaScript best practices is followed.

### CODE REVIEW

File: src/redux/categories/categories.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const STATUS_CHECKED = 'STATUS_CHECKED';
+
+export const statusChecked = () => ({
+  STATUS_CHECKED,
+  status: true,
```
Comment (GzimAsani) [inline src/redux/categories/categories.js]: - [x] The status should return `under construction`

File: src/components/Books.js [hunk 2]
```diff
@@ -1,19 +1,24 @@
+/* eslint-disable import/prefer-default-export */
+/* eslint-disable react/prop-types */
```
Comment (GzimAsani) [inline src/components/Books.js:2]: - [ ] Please do not disable linters, they are there to help you write better code. Do not do that unless you have a strong reason which prevents you from achieving the result

File: src/redux/books/books.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+export default function books(prevState = [], action) {
+  const index = prevState.indexOf(action.book);
+  switch (action.type) {
+    case BOOK_ADDED:
+      return [...prevState, action.book];
+    case BOOK_REMOVED:
+      return prevState
+        .slice(0, index)
+        .concat(prevState.slice(index + 1, prevState.length));
```
Comment (GzimAsani) [inline src/redux/books/books.js]: - [ OPTIONAL] I would suggest that you use here `filter` in order to remove the book

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi  @mnekx           
Good job so far! You are almost done :+1:  :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Good job with the git-flow :+1:
- Good PR description :clap: 

## Required Changes ♻️ 
- [x] _Check the comments below :arrow_double_down: _

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Books.js (hunk 2 above)
Comment (mnekx) [inline src/components/Books.js]: Could you show me how to resolve this error please, I tried destructuring with `{books}=props` and yet the error persists. How should I code this better? see below code screenshot

image

File: src/components/Book.js [hunk 4]
```diff
@@ -1,19 +1,26 @@
+/* eslint-disable react/prop-types */
```
Comment (TedLivist) [inline src/components/Book.js:1]: - [x] **[OPTIONAL]** For the `prop` issue you were having, let me walk you through how you can resolve it.

  - Install the `prop-types` package by running the command: `npm i prop-types`
  - Remove this eslint-disabling here  
  - In the top of this file, import it like so: `import PropTypes from 'prop-types';`
  - Towards the end of your file, validate your props like so:
    ```
    Book.propTypes = {
      book: PropTypes.shape({
        id: PropTypes.string,
        author: PropTypes.string,
        title: PropTypes.string,
      }).isRequired,
    };
    ```

  You can refer to this React documentation article about [Typechecking with proptypes](https://reactjs.org/docs/typechecking-with-proptypes.html)

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @mnekx,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Book.js (hunk 4 above)
Comment (mnekx) [reply src/components/Book.js:1]: Thank you very much, man. I appreciate it 👍

**Pull request merged**